### ISSUE TITLE
[BUG]: `no-duplicate-hooks` does not account for for-loops

### ISSUE THREAD

**Issue by cpAdm** [mskelton/eslint-plugin-playwright#459]
Given the following test file:

(based on example in official docs: https://playwright.dev/docs/test-parameterize)

```ts
import { expect, test } from '../utils/test';

[
	{ name: 'Alice', expected: 'Hello, Alice!' },
	{ name: 'Bob', expected: 'Hello, Bob!' },
	{ name: 'Charlie', expected: 'Hello, Charlie!' },
].forEach(({ name, expected }) => {
	test.describe(() => {
		test.beforeEach(async ({ page }) => {
			await page.goto(`https://example.com/greet?name=${name}`);
		});

		test(`testing with ${expected}`, async ({ page }) => {
			await expect(page.getByRole('heading')).toHaveText(expected);
		});
	});
});

[
	{ name: 'Charlie', expected: 'Hello, Charlie!' },
	{ name: 'David', expected: 'Hello, David!' },
].forEach(({ name, expected }) => {
	test.describe(() => {
		test.beforeEach(async ({ page }) => {
			await page.goto(`https://example.com/greet2?name=${name}`);
		});

		test(`testing 2 with ${expected}`, async ({ page }) => {
			await expect(page.getByRole('heading')).toHaveText(expected);
		});
	});
});
```

Then the second `test.beforeEach` has an ESLint warning:

`ESLint: Duplicate beforeEach in describe block (playwright/no-duplicate-hooks)`

While expected no warnings, since the hooks are all in separate blocks.
